versions/development/SPEC.md

```diff
@@ -179,7 +179,7 @@ task hello {
   }
 
   runtime {
-    docker: "broadinstitute/my_image"
+    container: "docker://dbroadinstitute/my_image"
```

Comment:
Just a thought, maybe "encoding into a single string" is not the nicest way to do this any more, especially if if different technologies might use different formats for their strings?

Eg why not have a standard map format like:
```wdl
runtime {
 container: {
  "organization": "broadinstitute"
  "image_name": "myimage"
  "floating_tag": "latest"
  "technologies": ["singularity", "docker", "foocontainertech"]
 }
}
```

The nice thing about this is that the WDL author doesn't need to worry about different string formatting styles in docker, singularity and foocontaintertech (which uses `\` instead of `/`).

Reply:
I get the sentiment here but do you really see people willingly typing all of that out?

Reply:
Isn't that something that can be handled by the execution engine given the single string format gives all the necessary information?
Furthermore a container string should always refer to a *single* image. Otherwise it is not reproducible. Given that it should refer to a single image, a single identifier string should be sufficient. Also more convenient. But I agree how this single string should like is still up for debate.